File: OCTOBER-2025/24-10-2025/p2048.java
class Solution {
    public boolean isBeautiful(int num){
        int[] count = new int[10];
        while(num > 0){
            int digit = num % 10;
            count[digit]++;
            num = num / 10;
        }
        for(int i=0;i<10;i++){
            if(count[i] != 0 && count[i] != i)
                return false;
        }
        return true;
    }

    public int nextBeautifulNumber(int n) {
        for(int i=n+1;i<=1666666;i++){
            if(isBeautiful(i))
                return i;
        }  
        return 0;
    }
}

File: OCTOBER-2025/24-10-2025/p2048.cpp
class Solution {

public:
    int nextBeautifulNumber(int n) {
        while(true){
            n++;
            vector<int>freq(10,0);


            int temp =n;
            while(temp){
                freq[temp%10]++;
                temp/=10;
            }


             bool isBalanced = true;
             for(int i=0 ; i<10 ; i++){
                if(freq[i]!=0 && freq[i]!=i){
                    isBalanced=false;
                    break;
                }
             }

              if (isBalanced) return n; 

        }
    }
};